src/cryptsetup/cryptsetup-generator.c

```diff
@@ -49,20 +49,20 @@ static int generate_keydev_mount(const char *name, const char *keydev, char **un
         assert(mount);
 
         r = mkdir_parents("/run/systemd/cryptsetup", 0755);
-        if (r < 0)
+        if (r < 0 && r != -EEXIST)
                 return r;
 
         r = mkdir("/run/systemd/cryptsetup", 0700);
-        if (r < 0)
-                return r;
+        if (r < 0 && errno != EEXIST)
+                return -errno;
 
         where = strjoin("/run/systemd/cryptsetup/keydev-", name);
```

Comment:
btw, is `name` known to be safe to be included in filenames? i don't think it is. there's some escaping necessary for this, no?

Reply:
Usually, it is luks-<UUID> however it could be something else if a user used ```rd.luks.name``` option on the kernel command line. You are right, we should escape the name argument.